### PR TITLE
Adding support for custom eval configs

### PR DESCRIPTION
**Pull request by yasser-sulaiman**
by adding include_path to EleutherAI specific eval args section in eleuther_evaluation.yaml file, custom lm-eval file can be used

#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.
solves #1054 

#### Changelog
What are the changes made in this PR?

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help.)

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add unit tests for any new functionality
- [ ] update docstrings for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
	- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

**Comment by ebsmothers** [comment]
@yasser-sulaiman thanks for the PR! Overall the changes look good to me. Can you update the test plan with a sanity check demonstrating that you're now able to load a custom task by passing `include_path`?

**Comment by yasser-sulaiman** [comment]
> @yasser-sulaiman thanks for the PR! Overall the changes look good to me. Can you update the test plan with a sanity check demonstrating that you're now able to load a custom task by passing `include_path`?

Hello, 
I am not sure I understood it, should I include an example demonstrating how it works or should I add a unit test for it?

**Comment by joecummings** [comment]
> > @yasser-sulaiman thanks for the PR! Overall the changes look good to me. Can you update the test plan with a sanity check demonstrating that you're now able to load a custom task by passing `include_path`?
> 
> 
> 
> Hello, 
> 
> I am not sure I understood it, should I include an example demonstrating how it works or should I add a unit test for it?

A code example demonstrating it works as expected (in the form of a Gist) is good.

**Comment by yasser-sulaiman** [comment]
> > > @yasser-sulaiman thanks for the PR! Overall the changes look good to me. Can you update the test plan with a sanity check demonstrating that you're now able to load a custom task by passing `include_path`?
> > 
> > 
> > Hello,
> > I am not sure I understood it, should I include an example demonstrating how it works or should I add a unit test for it?
> 
> A code example demonstrating it works as expected (in the form of a Gist) is good.

thanks for the suggestion, [here](https://gist.github.com/yasser-sulaiman/91ef64a428e1063064c6e556ec5833bd) is the link to an example configuration that demonstrates it works as expected

File: recipes/eleuther_eval.py [hunk 1]
```diff
@@ -241,7 +241,12 @@ def evaluate(self) -> None:
         except Exception:
             pass
 
-        task_dict = get_task_dict(self._tasks)
+        if self._cfg.get("include_path", False):
```
Comment (RdoubleA) [inline recipes/eleuther_eval.py:244]: can we do something like this?
```
task_manager = TaskManager(include_path=self._cfg.get("include_path", None))
task_dict = get_task_dict(self._tasks, task_manager)
```

**Review by RdoubleA — COMMENTED** [review]
Thanks for providing the code snippet! Let's actually update the eleuther eval recipe test to test this functionality. Maybe we can just check that the TaskManager is created correctly and eval runs successfully. Also left one minor comment

File: recipes/eleuther_eval.py (hunk 1 above)
Comment (yasser-sulaiman) [reply recipes/eleuther_eval.py]: I applied the minor comment but did not know how to write a test case for it :(

**Review by RdoubleA — APPROVED** [review]

**Pull request merged**